plugins/restful/RestfulEntityBase.php

```diff
@@ -640,6 +640,11 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $null_missi
       }
 
       $property_name = $info['property'];
+      $sub_property_name = FALSE;
+      // Set sub-property name, in case of nid bypass, since is is not required.
+      if (isset($info['sub_property']) && $info['sub_property'] != 'nid') {
```

Comment:
When are having `'nid'` as a _sub\_property_?

Also, this `'nid'` logic wouldn't scale for other entity types. Maybe consider using `entity_get_info($this->getEntityType())` to get the entity id key.